src/base/rss/rss_autodownloader.cpp

```diff
@@ -245,6 +245,7 @@ void AutoDownloader::processJob(const QSharedPointer<ProcessingJob> &job)
         }
 
         rule.setLastMatch(job->articleDate);
+        rule.matched(job->articleTitle);
```

Comment:
There is some incorrect logic. Why is `computeEpisodeName()` called twice for the same article?
Here is my solution:
1. Create `mutable` field in **AutoDownloadRuleData** struct named *lastEpisodeName*
2. Store last episode name inside `AutoDownloadRule::matches()`
3. Create the following method (instead of your `AutoDownloadRule::matched()`):
```c++
void AutoDownloadRule::storeLastMatch(const QDateTime &matchTime)
{
    setLastMatch(matchTime);
    
    if (useSmartFilter() && !m_dataPtr->lastEpisodeName.isEmpty()) {
        // TODO: probably need to add a marker for PROPER/REPACK to avoid duplicate downloads
        m_dataPtr->previouslyMatchedEpisodes.append(m_dataPtr->lastEpisodeName);
        m_dataPtr->lastEpisodeName.clear();
    }
}
```
4. Call `rule.storeLastMatch(job->articleDate);` instead of `rule.setLastMatch(job->articleDate); rule.matched(job->articleTitle);`

What do you say?